geoh5py/objects/surveys/direct_current.py

```diff
@@ -115,119 +132,94 @@ def ab_map(self) -> ReferenceValueMap | None:
-            mask[complement.cells[cell_mask, :]] = True
-
-            new_complement = super(Curve, complement).copy(  # type: ignore
-                parent=parent,
-                omit_list=omit_list,
-                copy_children=copy_children,
-                clear_cache=clear_cache,
-                mask=mask,
-                cell_mask=cell_mask,
+            raise AttributeError(
+                "The 'ab_cell_id' must be set on both the object and its complement."
             )
 
-            if isinstance(self, PotentialElectrode):
-                new_entity.current_electrodes = new_complement
-            else:
-                new_entity.potential_electrodes = new_complement
-
-            if new_complement.ab_cell_id is None and complement.ab_cell_id is not None:
-                new_complement.ab_cell_id = complement.ab_cell_id.values[cell_mask]
-
-            # Re-number the ab_cell_id
-            value_map = {
-                val: ind
-                for ind, val in enumerate(
-                    np.r_[0, np.unique(new_entity.current_electrodes.ab_cell_id.values)]
-                )
-            }
-            new_map = {
-                val: dict(new_entity.current_electrodes.ab_cell_id.value_map.map)[val]
-                for val in value_map.values()
-            }
-            new_complement.ab_cell_id.values = np.asarray(
-                [value_map[val] for val in new_complement.ab_cell_id.values]
-            )
-            new_entity.ab_cell_id.values = np.asarray(
-                [value_map[val] for val in new_entity.ab_cell_id.values]
+        # Re-number the ab_cell_id
+        value_map = {
+            val: ind
+            for ind, val in enumerate(
+                np.r_[0, np.unique(self.current_electrodes.ab_cell_id.values)]
             )
-            new_entity.ab_cell_id.entity_type.value_map = new_map
+        }
+        new_map = {
+            val: dict(self.current_electrodes.ab_cell_id.value_map.map)[val]
+            for val in value_map.values()
+        }
+        self.complement.ab_cell_id.values = np.asarray(
+            [value_map[val] for val in self.complement.ab_cell_id.values]
+        )
+        self.ab_cell_id.values = np.asarray(
+            [value_map[val] for val in self.ab_cell_id.values]
+        )
+        self.ab_cell_id.entity_type.value_map = new_map  # type: ignore
 
-        return new_entity
+    @property
+    def omit_list(self) -> tuple:
+        """
+        List of attributes to omit when copying.
+        """
+        return OMIT_LIST
 
     @property
     @abstractmethod
-    def current_electrodes(self):
+    def potential_electrodes(self):
         """
-        The associated current_electrodes (transmitters)
+        The associated potential_electrodes (receivers)
         """
 
+    @property
+    def type_map(self) -> dict[str, str]:
+        """
+        Mapping of the electrode types to the associated electrode.
+        """
+        return TYPE_MAP
```

Comment:
could be pertinent to make them class attributes are they are associated to the class.